server/src/main/java/org/apache/kafka/server/share/session/ShareSessionCache.java

```diff
@@ -90,6 +90,14 @@ public synchronized int size() {
         return sessions.size();
     }
 
+    /**
+     * Remove all the share sessions from cache.
+     */
+    public synchronized void removeAllSessions() {
+        sessions.clear();
+        numPartitions = 0;
```

Comment:
I observe that the `connectionIdToSessionMap` is not clearer. I think that's harmless and it will gradually empty as the connections close.